### PR TITLE
Add supported platforms document

### PR DESCRIPTION
**Pull request by hasheddan**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

Adds the initial documentation on supported platforms for each
Kubernetes component. This document will continue to be iterated upon as
procedures for supporting / dropping support for platforms becomes more
defined. It will also be updated with tiers of support and current test
coverage.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

#### Which issue(s) this PR fixes:

xref: https://github.com/kubernetes/sig-release/issues/1337

#### Special notes for your reviewer:

/assign @saschagrunert @justaugustus 
/area release-eng

### CODE REVIEW

**Review by justaugustus — CHANGES_REQUESTED** [review]
@hasheddan -- Very awesome!

Two things:
- As we start to draw the line between what is released and what is supported, I think it's important to put a disclaimer here, something in the way of "We test on linux/amd64 and that's the only thing we're 'really sure' about. Everything else is best effort."
- Can you break these out into two tables: server and client

**Comment by dims** [comment]
/hold

for reviews! :)

File: supported-platforms.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Supported Platforms
```
Comment (saschagrunert) [inline supported-platforms.md:1]: Do we want to put this into `/release-engineering`? And link it from the `README.md` there?

File: supported-platforms.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+A Kubernetes release consists of a set of Go binaries for various Operating Systems and CPU architectures. Platforms are identified using the Go `<GOOS>-<GOARCH>` parlance.
```
Comment (saschagrunert) [inline supported-platforms.md:3]:
```suggestion
A Kubernetes release consists of a set of Go binaries and container images for various Operating Systems and CPU architectures. Platforms are identified using the Go `<GOOS>-<GOARCH>` parlance.
```

File: supported-platforms.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Released Binaries
```
Comment (saschagrunert) [inline supported-platforms.md:5]: I think we can map this 1:1 to the container images, right? I guess we just have to mention this fact here, too.

File: supported-platforms.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+| kube-controller-manager |              |           |      X      |     X     |      X      |       X       |      X      |             |               |
+|        kube-proxy       |              |           |      X      |     X     |      X      |       X       |      X      |             |       X       |
+|      kube-scheduler     |              |           |      X      |     X     |      X      |       X       |      X      |             |               |
+|         kubeadm         |              |           |      X      |     X     |      X      |       X       |      X      |             |       X       |
+|         kubectl         |       X      |     X     |      X      |     X     |      X      |       X       |      X      |      X      |       X       |
+|         kubelet         |              |           |      X      |     X     |      X      |       X       |      X      |             |       X       |
+|         mounter         |              |           |      X      |     X     |      X      |       X       |      X      |             |               |
```
Comment (saschagrunert) [inline supported-platforms.md:20]: As a follow-up, my idea would be that we put this matrix in a map in k/release and verify it during the release cut. WDYT?

**Review by saschagrunert — COMMENTED** [review]
Thank you! Just some thoughts on container images, otherwise LGTM :point_down:

File: supported-platforms.md (hunk 4 above)
Comment (hasheddan) [reply supported-platforms.md:20]: @saschagrunert definitely :) I have an item on #1337 to do that :+1:

File: supported-platforms.md (hunk 3 above)
Comment (hasheddan) [reply supported-platforms.md:5]: Yep that sounds good :+1:

File: supported-platforms.md (hunk 1 above)
Comment (hasheddan) [reply supported-platforms.md:1]: So I was thinking about that, but I think as this document continues to mature, it actually is generally applicable to sig-release, similar to how the release-blocking / release-informing doc is at root of repo.

File: supported-platforms.md (hunk 3 above)
Comment (hasheddan) [reply supported-platforms.md:5]: @saschagrunert in thinking about this more, the container images we publish don't really map 1:1 here, so I think it makes sense to maybe follow-up with some additional documentation on that

File: supported-platforms.md (hunk 2 above)
Comment (spiffxp) [reply supported-platforms.md:3]: There are also various scripts and manifests, looking at the contents of e.g. https://github.com/kubernetes/kubernetes/releases/download/v1.18.12/kubernetes.tar.gz

Comment (hasheddan) [reply supported-platforms.md:3]: @spiffxp yep, this is not meant to cover _every artifact_ produced, just the supported platforms. That being said, I do think we should document those as well.

**Comment by hasheddan** [comment]
So I completely missed that an excellent document that describes what I have here as well as much more already exists (https://github.com/kubernetes/sig-release/blob/master/release-engineering/artifacts.md). I am going to close this out and do some work on cleaning up the existing doc :+1: 

/close

**Pull request closed without merging**